compiler/src/iree/compiler/Codegen/Common/TransformExtensions/CommonExtensionsOps.td

```diff
@@ -462,6 +462,28 @@ def ShareForallOperandsOp : Op<
   }];
 }
 
+def LowerSingleThreadedWorkgroupCountRegionOp :
+    Op<Transform_Dialect, "iree.lower_single_threaded_workgroup_count_region_op",
```

Comment:
Shouldn't this be more like "single_workgroup_distribution" or something like that to state that it's not just a single thread executing but also that there is a single workgroup? (i.e., a single thread can execute multiple workgroups, which is what we were doing in the experiments by default).

Reply:
Also wondering if we need a dedicated op for the single-workgroup lowering or we could just have one op that invokes TileAndFuse and then call it with tile sizes 0, 0, 0..., 0 to get no distribution and would also let us play with other flavors of distribution by just providing different tile sizes. WDYT?

Reply:
Good point, I'll update the op name.

> ... call it with tile sizes 0, 0, 0..., 0 to get no distribution and would also let us play with other flavors of distribution by just providing different tile sizes.

We're not able for doing it in transform dialect. Because they are not using TileAndDistributeToWorkgroups pass.  The existing transform op distributes them to scf.for_all and lower flow.workgroup_count* ops. What we need is a single pattern/transform that lowers workgroup_count ops to `c1, c1, c1`. See the lit test for an example.

Reply:
I'm missing context but, regardless of the distribution construct/lowering, there should be a way to create a single workgroup that covers all the iteration space with the existing transfor dialect ops, isn't that the case? What am I missing?

Reply:
They are all merged into a single operation, AFAIK. This is why I'm creating a new op for doing it . Happy to discuss more in our 1:1 😀